kobo/apps/superuser_stats/views.py

```diff
@@ -65,16 +261,59 @@ def user_report(request):
         base_filename, request.user.username)
     generate_user_report.delay(filename)
     template_ish = (
-        '<html><head><title>Hello, superuser.</title></head>'
-        '<body>Your report is being generated. Once finished, it will be '
-        'available at <a href="{0}">{0}</a>. If you receive a 404, please '
-        'refresh your browser periodically until your request succeeds.'
-        '</body></html>'
-    ).format(base_filename)
+        f'<html><head><title>Users report</title></head>'
+        f'<body>Your report is being generated. Once finished, it will be '
+        f'available at <a href="{base_filename}">{base_filename}</a>.<br>'
+        f'If you receive a 404, please refresh your browser periodically until '
+        f'your request succeeds.'
+        f'</body></html>'
+    )
     return HttpResponse(template_ish)
 
 
 @user_passes_test(lambda u: u.is_superuser)
+def user_statistics_report(request):
+    """
+    View for the User Statisctics Report
+    Generates a detailed report of a user's activity
+    over a period of time
+    """
+    base_filename = 'user-statistics-report_{}_{}_{}.csv'.format(
+        re.sub('[^a-zA-Z0-9]', '-', request.META['HTTP_HOST']),
+        date.today(),
+        datetime.now().microsecond
+    )
+
+    # Get the date filters from the query and set defaults
+    today = timezone.now().date()
+    start_date = request.GET.get('start_date', today)
+    tomorrow = today + timedelta(days=1)
+    end_date = request.GET.get('end_date', tomorrow)
+
+    # Generate the CSV file
+    filename = _base_filename_to_full_filename(
+        base_filename, request.user.username)
+    generate_user_statistics_report.delay(filename, start_date, end_date)
```

Comment:
This also doesn't appear to be working correctly as I'm only presented with `Not Found. The requested resource was not found on this server.` even after a while waiting and refreshing.

Reply:
Maybe this should also make sure to handle the different data types this field can hold, grabbing the country code from the list of country dicts)

Current:
Username | Date Joined | Organization | Country
-- | -- | -- | --
joshuaberetta | 2021-07-07 18:38:10.161650+00:00 | Kobo | [{'label': 'Canada', 'value': 'CAN'}, {'label': 'South Africa', 'value': 'ZAF'}]

Better:
Username | Date Joined | Organization | Country
-- | -- | -- | --
joshuaberetta | 2021-07-07 18:38:10.161650+00:00 | Kobo | CAN, ZAF